### PR TITLE
Reviewed typos, comments, instructions, etc.

### PR DESCRIPTION
**Pull request by pires**
/cc @errordeveloper @lukemarsden

### CODE REVIEW

**Comment by luxas** [comment]
LGTM, thanks!

File: cmd/kubeadm/app/api/types.go [hunk 1]
```diff
@@ -77,20 +78,14 @@ func init() {
 	}
 }
 
+// JoinFlags holds values for "kubeadm join" command flags.
 type JoinFlags struct {
 	MasterAddrs []net.IP
 }
 
-// TODO(phase1?) we haven't decided whether manual sub commands should get merged into  main commands...
-type ManualFlags struct {
```
Comment (errordeveloper) [inline cmd/kubeadm/app/api/types.go:85]: Thanks for removing it, I've been meaning to do it.

File: cmd/kubeadm/app/cmd/cmd.go [hunk 2]
```diff
@@ -51,14 +51,14 @@ func NewKubeadmCommand(f *cmdutil.Factory, in io.Reader, out, err io.Writer, env
 			    ┌──────────────────────────────────────────────────────────┐
 			    │  On the first machine                                    │
 			    ├──────────────────────────────────────────────────────────┤
-			    │ master# kubeadm init master                              │
+			    │ master# kubeadm init                                     │
 			    │ Your token is: <token>                                   │
```
Comment (errordeveloper) [inline cmd/kubeadm/app/cmd/cmd.go]: This isn't the format of the output we have ended-up with, what do you think we could do here? May be just remove the line?

File: cmd/kubeadm/app/kubeadm.go [hunk 3]
```diff
@@ -56,7 +54,6 @@ func getEnvParams() map[string]string {
 }
 
 func Run() error {
-	CommandLine = pflag.NewFlagSet(os.Args[0], pflag.ContinueOnError)
```
Comment (errordeveloper) [inline cmd/kubeadm/app/kubeadm.go:59]: You are right, this actually didn't do anything. I've added it when I was trying to get rid of the flags which leaked from some other pkgs, but now hide those explicitly and it works well.

**Review by errordeveloper — CHANGES_REQUESTED** [review]
LGTM, but could please also remove word server from short descriptions?

```
cmd/kubeadm/app/cmd/init.go:            Short: "Run this on the first server you deploy onto.",
cmd/kubeadm/app/cmd/join.go:            Short: "Run this on other servers to join an existing cluster.",
```

Let's replace it with "machine", which has no implication about people using VMs vs bare-metal, or any other kind of devices.

File: cmd/kubeadm/app/cmd/cmd.go (hunk 2 above)
Comment (pires) [reply cmd/kubeadm/app/cmd/cmd.go]: Yes, I'll remove the output lines as it may become hard to manage in the future.

**Comment by pires** [comment]
Change requests addressed.

File: cmd/kubeadm/README.md [hunk 4]
```diff
@@ -5,42 +5,63 @@
...
 It's usually enough to run `kubeadm init`, but in some case you might like to override the
-default behaviour.
+default behaviour. The flags used for said purpose are described below.
 
-- `--token=<str>`
+- `--token=<token>`
 
-By default, a token is generated, but if you are to automate cluster deployment, you want to
+By default, a token is generated, but if you are to automate cluster deployment, you will want to
 set the token ahead of time. Read the docs for more information on the token format.
```
Comment (lukemarsden) [inline cmd/kubeadm/README.md:13]: Which docs, where?

File: cmd/kubeadm/README.md [hunk 5]
```diff
@@ -5,42 +5,63 @@
...
+
+- `--schedule-workload=<bool>` (default: "false")
+
+By default, `kubeadm` sets the master node kubelet as non-schedulable for workloads. This means the master node won't run your pods. If you want to change that, 
+use `--schedule-workload=true`.
+
+- `--cloud-provider=<cloud provider>`
+
+By default, `kubeadm` doesn't perform auto-detection of the current cloud provider. If you want to specify it, use `--cloud-provider`. Possible values are
+the ones supported by controller-manager, namely `"aws"`, `"azure"`, `"cloudstack"`, `"gce"`, `"mesos"`, `"openstack"`, `"ovirt"`, `"rackspace"`, `"vsphere"`.
```
Comment (lukemarsden) [inline cmd/kubeadm/README.md:40]: What happens if I set this value? Do I need to do anything special with IAM roles? This needs to be explained here, or we need to link to some documentation on the matter.

File: cmd/kubeadm/README.md [hunk 6]
```diff
@@ -5,42 +5,63 @@
...
 
 ***TODO(phase1+)***
```
Comment (lukemarsden) [inline cmd/kubeadm/README.md:42]: Can we move these TODOs into an issue?

File: cmd/kubeadm/README.md [hunk 7]
```diff
@@ -5,42 +5,63 @@
...
 - `--api-bind-loopback-unsecure=<bool>`
 
-***TODO(pahse2)***
-
 - `--prefer-private-network=<bool>`
 - `--prefer-public-network=<bool>`
 
 ### `kubeadm join`
 
+`kubeadm join` has one mandatory flag, the token used to secure cluster bootstrap, and one mandatory argument, the master IP address.
```
Comment (lukemarsden) [inline cmd/kubeadm/README.md:56]: presumably DNS name would work too? Can we just say "IP address or hostname"?

File: cmd/kubeadm/README.md [hunk 8]
```diff
@@ -5,42 +5,63 @@
...
+Here's an example on how to use it:
+
+`kubeadm join --token=the_secret_token 192.168.1.1`
+
+- `--token=<token>`
+
+By default, when `kubeadm init` runs, a token is generated and revealed in the output. That's the token you should use here.
+
 # User Experience Considerations
```
Comment (lukemarsden) [inline cmd/kubeadm/README.md:65]: Let's move this whole section into an issue or a design proposal or something.

File: cmd/kubeadm/app/cmd/init.go [hunk 9]
```diff
@@ -32,20 +32,21 @@ import (
 )
 
 var (
-	init_done_msgf = dedent.Dedent(`
+	initDoneMsgf = dedent.Dedent(`
 		Kubernetes master initialised successfully!
 
-		You can connect any number of nodes by running:
+		You can now join any number of machines by running the following on each node:
```
Comment (lukemarsden) [inline cmd/kubeadm/app/cmd/init.go:38]: "You can now join any number of machines as nodes by running the following on each of them" is clearer IMO. Although the user may not understand what a "node" is...

File: cmd/kubeadm/app/cmd/init.go [hunk 10]
```diff
@@ -54,28 +55,28 @@ func NewCmdInit(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
...
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.Services.DNSDomain, "service-dns-domain", kubeadmapi.DefaultServiceDNSDomain,
-		`(optional) use alterantive domain name for services, e.g. "myorg.internal"`,
+		`(optional) use alternative domain for services, e.g. "myorg.internal"`,
 	)
 	cmd.PersistentFlags().StringVar(
 		&s.InitFlags.CloudProvider, "cloud-provider", "",
-		`(optional) enable cloud proiver features (external load-balancers, storage, etc)`,
+		`(optional) enable a specific cloud provider features (external load-balancers, storage, etc), e.g. "gce"`,
```
Comment (lukemarsden) [inline cmd/kubeadm/app/cmd/init.go:79]: s/a //

Comment (lukemarsden) [reply cmd/kubeadm/app/cmd/init.go:79]: Again, would be nice if this linked to a doc. I don't actually know how this bit works atm though so I'm struggling to document it :)

File: cmd/kubeadm/app/cmd/join.go [hunk 11]
```diff
@@ -31,35 +31,36 @@ import (
...
 var (
-	join_done_msgf = dedent.Dedent(`
+	joinDoneMsgf = dedent.Dedent(`
 		Node join complete:
 		* Certificate signing request sent to master and response
 		  received.
 		* Kubelet informed of new secure connection details.
 
-		Run 'kubectl get nodes' on the master to see this node join.
+		Run 'kubectl get nodes' on the master to see this machine join.
```
Comment (lukemarsden) [inline cmd/kubeadm/app/cmd/join.go:40]: well, the machine "becomes" a node when it joins... so maybe this was OK before? Also the `kubectl get nodes` command refers to nodes...

File: cmd/kubeadm/app/cmd/init.go [hunk 12]
```diff
@@ -32,20 +32,21 @@ import (
...
 		`)
 )
 
+// NewCmdInit returns "kubeadm init" command.
 func NewCmdInit(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 	advertiseAddrs := &[]string{}
 	cmd := &cobra.Command{
 		Use:   "init",
-		Short: "Run this on the first server you deploy onto.",
+		Short: "Run this on the first machine.",
```
Comment (lukemarsden) [inline cmd/kubeadm/app/cmd/init.go:49]: Please can you add a `Long` description here, which explains what will happen when you run this command?

File: cmd/kubeadm/app/cmd/join.go [hunk 13]
```diff
@@ -31,35 +31,36 @@ import (
...
 		`)
 )
 
+// NewCmdJoin returns "kubeadm join" command.
 func NewCmdJoin(out io.Writer, s *kubeadmapi.KubeadmConfig) *cobra.Command {
 	cmd := &cobra.Command{
 		Use:   "join",
-		Short: "Run this on other servers to join an existing cluster.",
+		Short: "Run this on any machine you wish to join an existing cluster.",
```
Comment (lukemarsden) [inline cmd/kubeadm/app/cmd/join.go:48]: Please can you add a `Long` description here, which explains what will happen when you run this command?

**Review by lukemarsden — COMMENTED** [review]
Thanks so much for this PR @pires! It's a big step forwards. Please could you address the few comments I left and then once @errordeveloper's happy with it we can get it merged. Thanks again!

File: cmd/kubeadm/README.md (hunk 4 above)
Comment (errordeveloper) [reply cmd/kubeadm/README.md:13]: Well, this was only meant to be a draft!

File: cmd/kubeadm/README.md (hunk 5 above)
Comment (errordeveloper) [reply cmd/kubeadm/README.md:40]: Yes, sure. But I think it's a bigger problem and we cannot explain it in this PR. I've opened https://github.com/kubernetes/kubernetes/issues/33037.

File: cmd/kubeadm/README.md (hunk 6 above)
Comment (errordeveloper) [reply cmd/kubeadm/README.md:42]: All todos will be moved into issues once the code is in master.

File: cmd/kubeadm/README.md (hunk 7 above)
Comment (errordeveloper) [reply cmd/kubeadm/README.md:56]: Not yet. Right now the code assumes IP address.

File: cmd/kubeadm/app/cmd/join.go (hunk 11 above)
Comment (errordeveloper) [reply cmd/kubeadm/app/cmd/join.go:40]: True, this shouldn't have changed.

**Review by errordeveloper — APPROVED** [review]

**Comment by errordeveloper** [comment]
@lukemarsden I'll merge it as is, and will take another pass tomorrow to address some of  your comments.

**Pull request merged**